apps/libsignpost/signpost_api.h

```diff
@@ -261,23 +265,39 @@ int signpost_processing_reply(uint8_t src_addr, uint8_t message_type, uint8_t* r
 /* ENERGY API                                                             */
 /**************************************************************************/
 
+//message types
 enum energy_message_type {
     EnergyQueryMessage = 0,
     EnergyLevelWarning24hMessage = 1,
     EnergyLevelCritical24hMessage = 2,
     EnergyCurrentWarning60sMessage = 3,
+    EnergyReportMessage = 4,
+    EnergyDutyCycleMessage = 5,
 };
 
+//information sent to a module from the controller
 typedef struct __attribute__((packed)) energy_information {
-    uint32_t    energy_limit_24h_mJ;
-    uint32_t    energy_used_24h_mJ;
-    uint16_t    current_limit_60s_mA;
-    uint16_t    current_average_60s_mA;
+    uint32_t    energy_limit_mWh;
+    uint16_t    average_power_mW;
     uint8_t     energy_limit_warning_threshold;
     uint8_t     energy_limit_critical_threshold;
 } signpost_energy_information_t;
 
-_Static_assert(sizeof(signpost_energy_information_t) == 14, "On-wire structure size");
+_Static_assert(sizeof(signpost_energy_information_t) == 8, "On-wire structure size");
+
+//a mechanism for modules to report energy usage from other modules
+//this structure is for each module
+typedef struct __attribute__((packed)) energy_report_module {
+    uint8_t module_address; //module i2c address
+    uint8_t module_percent; //an integer percent 0-100 that  the module has used
```

Comment:
okay I'm late to the game on this one, but I'm really confused by this interface, I feel like it's missing really important information to attribute energy to other modules

  - What's the window of time that this "energy report" should be for? Since the last call? Does the mean if I start a radio operation I have to call this once with a null module before the op, then send the packet, then again after? (Or else if I'm the first packet in the morning I get charged with all the idle power draw of the radio?)
  - It feels way more likely that the radio would turn on to send a bunch of packets, I'd have expected this to be an array of percentages for all modules (with a num_modules in packet as well for generalizability), especially given the latency to send/reply i2c messages / measure energy, plus the high energy startup/shutdown/idle costs of radios, (the costs of which you want to amortize across all the modules that demanded a wakeup)

I don't think it makes a big difference whether the energy math is done on the radio or controller side